apps/nextjs/src/hooks/useTemporaryLessonPlanWithStreamingEdits.ts

```diff
@@ -55,90 +55,101 @@ export const useTemporaryLessonPlanWithStreamingEdits = ({
   partialPatches: PatchDocument[];
 } => {
   const throttledAssistantMessages = useThrottle(messages, 100);
-  const [tempLessonPlan, setTempLessonPlan] = useState<LooseLessonPlan>({
-    ...lessonPlan,
-  });
-  const [appliedPatches, setAppliedPatches] = useState<PatchDocumentWithHash[]>(
-    [],
-  );
+  const tempLessonPlanRef = useRef<LooseLessonPlan>({});
+  const appliedPatchesRef = useRef<PatchDocumentWithHash[]>([]);
 
+  // Update the ref when lessonPlan changes
   useEffect(() => {
-    if (!isStreaming && lessonPlan) {
-      if (!equals(lessonPlan, tempLessonPlan)) {
-        console.log("Reset the temp lesson plan to the lesson plan");
-        setTempLessonPlan(deepClone(lessonPlan));
-      }
+    if (lessonPlan && !equals(lessonPlan, tempLessonPlanRef.current)) {
```

Comment:
Doing this means that the lessonPlan object is stable, which results in the main component tree not doing a full re-render when we change the values of keys